src/scripting/lua_kernel_base.cpp

```diff
@@ -711,7 +711,7 @@ bool lua_kernel_base::protected_call(lua_State * L, int nArgs, int nRets, error_
 bool lua_kernel_base::load_string(char const * prog, const std::string& name, error_handler e_h)
 {
 	// pass 't' to prevent loading bytecode which is unsafe and can be used to escape the sandbox.
-	int errcode = luaL_loadbufferx(mState, prog, strlen(prog), name.empty() ? name.c_str() : prog, "t");
+	int errcode = luaL_loadbufferx(mState, prog, strlen(prog), name.empty() ? prog : name.c_str(), "t");
```

Comment:
This line looks serious enough to have a bug report and PR on its own. A unit test that would fail with the old code would also be good.

Reply:
It's not all that serious since this is just name shown in error messages, and the bug is just that our code doesn't use the name it's supposed to use.

But I can extract it into a separate commit, sure, and maybe add a unit test if I can think of a way to do it.

Reply:
I did manage to make a unit test for this, but it's reliant on this PR, so I don't think I can extract the unit test to a separate PR.

Reply:
Thanks for creating the test. Sorry for suggesting a separate PR for this change, I misunderstood what it did.

Reply:
To be clear, I meant the test is reliant on this PR, not that the change is. Even if I extracted the change to a separate PR, the test wouldn't be part of the PR.